### PR TITLE
TH v0.13.14, health check and migration improvements

### LINKED ISSUE #15: [bug]: Health check passes even though LND is not ready

### LINKED ISSUE #20: [chore]: Update dependency on LND v15

### LINKED ISSUE #21: [Chore]: Upstream update to TH v0.13.14

### PR DESCRIPTION
**Pull request by islandbitcoin**
- [x] #21 
- [x] #20 
- [x] #15

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,5 +28,4 @@ image.tar: Dockerfile docker_entrypoint.sh check-web.sh
 	DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build --tag start9/thunderhub/main:${VERSION} --platform=linux/arm64/v8 -o type=docker,dest=image.tar .
 
 scripts/embassy.js: scripts/**/*.ts
-	deno cache --reload scripts/embassy.ts
```
Comment (chrisguida) [inline Makefile:31]: 👍

File: scripts/deps.ts [hunk 2]
```diff
@@ -1 +1,2 @@
-export * from "https://deno.land/x/embassyd_sdk@v0.3.1.0.3/mod.ts";
\ No newline at end of file
+export * from "https://deno.land/x/embassyd_sdk@v0.3.1.1.1/mod.ts";
+export * from "https://deno.land/x/embassyd_sdk@v0.3.1.0.12/emvar-lite/mod.ts";
```
Comment (chrisguida) [inline scripts/deps.ts]:
```suggestion
export * from "https://deno.land/x/embassyd_sdk@v0.3.1.0.12/emvar-lite/mod.ts";

```

File: scripts/embassy.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
 export { setConfig } from "./services/setConfig.ts";
 export { properties } from "./services/properties.ts";
 export { getConfig } from "./services/getConfig.ts";
-export { dependencies } from "./services/dependencies.ts";
\ No newline at end of file
+export { migration } from "./services/migrations.ts";
```
Comment (chrisguida) [inline scripts/embassy.ts]:
```suggestion
export { migration } from "./services/migrations.ts";

```

File: scripts/services/migrations.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+          (config) => {
+            return config;
+          },
+          true,
+          { version: "0.13.14", type: "down" },
+        ),
+      },
+    },
+    "0.13.14",
+  );
```
Comment (chrisguida) [inline scripts/services/migrations.ts]:
```suggestion
  );

```

This migration doesn't appear to do anything.

File: scripts/services/setConfig.ts [hunk 5]
```diff
@@ -1,4 +1,12 @@
...
-export const setConfig = compat.setConfig;
\ No newline at end of file
+export const setConfig: T.ExpectedExports.setConfig = async (effects, input ) => {
+
+  const depsLnd: T.DependsOn = {lnd: ['synced']}
+
+  return await compat.setConfig(effects,input, {
+    ...depsLnd,
+  })
+}
```
Comment (chrisguida) [inline scripts/services/setConfig.ts]:
```suggestion
}

```

File: scripts/services/setConfig.ts [hunk 6]
```diff
@@ -1,4 +1,12 @@
 import {
   compat,
+  types as T
 } from "../deps.ts";
-export const setConfig = compat.setConfig;
\ No newline at end of file
+export const setConfig: T.ExpectedExports.setConfig = async (effects, input ) => {
+
+  const depsLnd: T.DependsOn = {lnd: ['synced']}
```
Comment (chrisguida) [inline scripts/services/setConfig.ts:7]: 👍

File: thunderhub.patch [hunk 7]
```diff
@@ -9,15 +9,15 @@
  # ---------------
  # Install Dependencies
  # ---------------
--FROM node:14.15-alpine as deps
-+FROM arm64v8/node:14.15-alpine as deps
+-FROM node:16.15.1-alpine as deps
++FROM arm64v8/node:16.15.1-alpine as deps
```
Comment (chrisguida) [inline thunderhub.patch:13]: what's the reason for this change?

File: thunderhub.patch [hunk 8]
```diff
@@ -9,15 +9,15 @@
...
  
  WORKDIR /app
  
  # Install dependencies neccesary for node-gyp on node alpine
  RUN apk add --update --no-cache \
    libc6-compat \
-   python \
+   python3 \
```
Comment (chrisguida) [inline thunderhub.patch:20]: I assume this is because TH has switched to python3 under the hood?

**Review by chrisguida — CHANGES_REQUESTED** [review]
See comments

File: thunderhub.patch (hunk 7 above)
Comment (islandbitcoin) [reply thunderhub.patch:13]: keeping the patch in sync with upstream

File: thunderhub.patch (hunk 8 above)
Comment (islandbitcoin) [reply thunderhub.patch:20]: same as above

File: thunderhub.patch (hunk 7 above)
Comment (chrisguida) [reply thunderhub.patch:13]: ah, so upstream updated to the `arm64v8` version of the image?

Comment (islandbitcoin) [reply thunderhub.patch:13]: no, the upstream version updated from node v14.15 to node v16.15.1

File: scripts/services/migrations.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        ),
+        down: compat.migrations.updateConfig(
+          (config) => {
+            return config;
+          },
+          true,
+          { version: "0.13.14", type: "down" },
+        ),
+      },
+    },
```
Comment (dr-bonez) [inline scripts/services/migrations.ts]: If no migration is necessary you can leave this empty

File: scripts/services/setConfig.ts [hunk 10]
```diff
@@ -1,4 +1,12 @@
...
+
+  return await compat.setConfig(effects,input, {
+    ...depsLnd,
```
Comment (dr-bonez) [inline scripts/services/setConfig.ts]: why destructure? why not pass it directly in?

**Review by dr-bonez — APPROVED** [review]

File: scripts/services/setConfig.ts (hunk 10 above)
Comment (islandbitcoin) [reply scripts/services/setConfig.ts]: @Blu-J ?

File: scripts/services/migrations.ts (hunk 9 above)
Comment (islandbitcoin) [reply scripts/services/migrations.ts]: done.

File: thunderhub.patch (hunk 7 above)
Comment (chrisguida) [reply thunderhub.patch:13]: I see, so why the switch to `arm64v8` then?

Comment (chrisguida) [reply thunderhub.patch:13]: Gotcha, reading patch files is confusing xD

Ok so the change from vanilla alpine to `arm64v8` alpine was already baked in.

File: scripts/services/migrations.ts (hunk 4 above)
Comment (chrisguida) [inline scripts/services/migrations.ts]: This migration doesn't appear to do anything, let's delete

**Review by chrisguida — CHANGES_REQUESTED** [review]

File: scripts/services/setConfig.ts (hunk 10 above)
Comment (Blu-J) [reply scripts/services/setConfig.ts]: Yes, in this case makes sense to just pass it in.
The original example you pulled from used 3 things.

**Review by Blu-J — APPROVED** [review]

File: scripts/services/migrations.ts (hunk 4 above)
Comment (islandbitcoin) [reply scripts/services/migrations.ts]: deleted

File: scripts/services/setConfig.ts (hunk 10 above)
Comment (islandbitcoin) [reply scripts/services/setConfig.ts]: so.... does this need to change now? or can I roll with it as is to push LND v15? I don't know how to pass it in directly.

Comment (chrisguida) [reply scripts/services/setConfig.ts]: @islandbitcoin @dr-bonez @Blu-J is this resolved?

File: scripts/services/setConfig.ts [hunk 11]
```diff
@@ -1,4 +1,12 @@
...
+  })
```
Comment (Blu-J) [inline scripts/services/setConfig.ts]:
```suggestion
  return await compat.setConfig(effects,input, depsLnd)
```

File: scripts/services/setConfig.ts (hunk 10 above)
Comment (islandbitcoin) [reply scripts/services/setConfig.ts]: Blu-J made the change and I committed it in this PR. closing this thread
